### PR TITLE
Remove tracks and activities

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
This will eventually implement track and activity removal, but right now only the data management is implemented. It also fixes updating activities when a track ID is changed, and handles a corner case if an activity uses a track more than once.

### CODE REVIEW

**Comment by mwoehlke-kitware** [comment]
Track and activity removal is now implemented.

File: web_external/reducers/index.js [hunk 1]
```diff
@@ -77,8 +77,13 @@ function app(state, action) {
...
             var annotationTypeContainer = state.annotationTypeContainer.change(action.payload.trackId, action.payload.newTrackId, action.payload.newCset3);
-            // to do change track id inside activity
-            return { ...state, ...{ annotationGeometryContainer, annotationTypeContainer, pendingSave: true, selectedTrackId: action.payload.newTrackId } };
+            var annotationActivityContainer = state.annotationActivityContainer.changeTrack(action.payload.trackId, action.payload.newTrackId);
+            return { ...state, ...{ annotationGeometryContainer, annotationTypeContainer, annotationActivityContainer, pendingSave: true, selectedTrackId: action.payload.newTrackId } };
+        case types.DELETE_TRACK:
+            var annotationGeometryContainer = state.annotationGeometryContainer.removeTrack(action.payload);
+            var annotationTypeContainer = state.annotationTypeContainer.remove(action.payload);
+            var annotationActivityContainer = state.annotationActivityContainer.removeTrack(action.payload);
+            return { ...state, ...{ annotationGeometryContainer, annotationTypeContainer, pendingSave: true, selectedTrackId: null } };
```
Comment (matthewma7) [inline web_external/reducers/index.js]: We probably need to set editingTrackId to null as well

File: web_external/reducers/index.js [hunk 2]
```diff
@@ -90,6 +95,9 @@ function app(state, action) {
         case types.CHANGE_ACTIVITY:
             var annotationActivityContainer = state.annotationActivityContainer.change(action.payload.activityId, action.payload.newActivityAct2, action.payload.newTimespan);
             return { ...state, ...{ annotationActivityContainer, pendingSave: true } };
+        case types.DELETE_ACTIVITY:
+            var annotationActivityContainer = state.annotationActivityContainer.remove(action.payload);
+            return { ...state, ...{ annotationActivityContainer, pendingSave: true } };
```
Comment (matthewma7) [inline web_external/reducers/index.js]: We should set selectedActivityId to null in the new state.

**Review by matthewma7 — APPROVED** [review]
The features works well now. Approving.

**Pull request merged**